index.js

```diff
@@ -216,14 +218,25 @@ function request(method, url, options, callback) {
 
   var responded = false;
 
-  var req = protocols[url.protocol.replace(/\:$/, '')].request({
+  var params = {
     host: url.hostname,
     port: url.port,
     path: url.path,
     method: method,
     headers: rawHeaders,
     agent: agent
-  }, function (res) {
+  }
+
+  if (options.proxy === null || options.proxy === undefined || options.proxy === '') {
+    options.proxy = 'http_proxy' in env ? env.http_proxy : false;
+  }
+
+  if (options.proxy) {
+    var proxyAgent = new proxyAgents[url.protocol.replace(/\:$/, '')](options.proxy);
+    params.agent = proxyAgent;
```

Comment:
That makes sense.  Lets solve this as part of `sync-request` by allowing you to pass a `proxy` option to `sync-request`, since passing an `agent` won't work there.